src/qaqc/01-organize_raw_data.Rmd

```diff
@@ -307,57 +324,64 @@ temperature_archery <- temperature_archery %>%
 
 ```
 
-For wonky data next to each other (defined by roll sd) - here we make some seasonal
-summaries to help with this
+Site visits: 
 
 ```{r}
-temperature_summary <- temperature_archery %>% 
-  group_by(y_q) %>% 
-  summarise(yq_sd = sd(p1, na.rm = T))
+# flag the start time of a site visit
+temperature_archery <- temperature_archery %>%
+  mutate(flag = if_else((!is.na(visit_comments)), 
+                if_else(is.na(flag), "Site visit",
+                                paste(flag, "Site visit", sep = "; ")),
+                        flag))
 ```
 
+For wonky data next to each other (defined by roll sd) - here we make some seasonal summaries to help with this:
+
 ```{r}
+temperature_summary <- temperature_archery %>% 
+  group_by(y_m) %>% 
+  summarise(ym_sd = sd(p1, na.rm = T))
+
 # flag data points that are outside the range of 3 sds from the mean of previous + future 3 data points:
 temperature_archery <- temperature_archery %>%
   left_join(temperature_summary) %>% 
-  mutate(flag = if_else((p1 <= rollavg - (3 * yq_sd) | p1 >= rollavg + (3 * yq_sd)),
-                        "Outside SD range", 
+  # Do we want to play around with using the monthly sd instead of the rolling sd?
+    mutate(flag = if_else((p1 <= rollavg - (3 * ym_sd) | p1 >= rollavg + (3 * ym_sd)), 
+                if_else(is.na(flag), "Outside SD range",
+                                paste(flag, "Outside sd range", sep = "; ")),
                         flag))
-
 ```
 
+For data that repeats:
+
 ```{r}
 # flag data that repeat across time:
 temperature_archery <- temperature_archery %>% 
   mutate(flag = if_else((p1 == front1 | p1 == back1), 
-                       if_else(is.na(flag),
-                               "Repeated value",
-                               paste(flag, "Repeated value", sep = "; ")),
-                       flag))
-
+                if_else(is.na(flag), "Repeated value",
+                                paste(flag, "Repeated value", sep = "; ")),
+                        flag))
 ```
 
+For instances where change in a 15m period is > 1degC
+
 ```{r}
 # flag data where change in a 15m period is > 1degC
 temperature_archery <- temperature_archery %>% 
   mutate(flag = if_else(slope_ahead >= slope_thresh | slope_behind >= slope_thresh, 
                         if_else(is.na(flag),
-                               "Slope exceedance",
-                               paste(flag, "Slope exceedance", sep = "; ")),
+                                "Slope exceedance",
```

Comment:
todo/suggestion: this is another spot that only tags the beginning and end of a run of data that are likely to be problematic and need to be flagged. The first data point where the slope ahead is greater than the threshold is *technically* an acceptable value. The flag for slope ahead should actually be the NEXT observation.